runtime/runtime/src/lib.rs

```diff
@@ -288,6 +289,35 @@ impl Runtime {
         debug!(target: "runtime", "{}", log_str);
     }
 
+    /// Parallel validation for all transactions.
+    ///
+    /// If any transaction fails, returns an error immediately. Otherwise, returns Ok(()).
+    fn parallel_validate_transactions(
+        config: &RuntimeConfig,
+        gas_price: Balance,
+        transactions: &[SignedTransaction],
+        current_protocol_version: ProtocolVersion,
+    ) -> Result<HashMap<CryptoHash, TransactionCost>, InvalidTxError> {
+        tracing::debug!(target: "runtime", "parallel validation: starting threads");
+
+        let results: Vec<(CryptoHash, Result<TransactionCost, InvalidTxError>)> = transactions
+            .par_iter()
+            .map(|tx| {
+                let res =
+                    validate_transaction(config, gas_price, tx, true, current_protocol_version);
+                (tx.get_hash(), res)
+            })
+            .collect();
```

Comment:
This collect is forcing the whole thing to be processed before any of the results are checked for `Err`. Maybe consider collecting into a `Result<Vec<...>>` which will terminate the iterator early?